adapter-base/src/main/java/org/eclipse/hono/adapter/AbstractProtocolAdapterApplication.java

```diff
@@ -372,16 +374,19 @@ protected void setCollaborators(final AbstractProtocolAdapterBase<?> adapter) {
                             protocolAdapterProperties.isJmsVendorPropsEnabled()));
         }
         if (!appConfig.isPubSubMessagingDisabled() && pubSubConfigProperties.isProjectIdConfigured()) {
-            LOG.info("Pub/Sub client configuration present, adding Pub/Sub messaging clients");
-
-            final var pubSubFactory = new CachingPubSubPublisherFactory(
-                    vertx,
-                    pubSubConfigProperties.getProjectId(),
-                    null);
-
-            telemetrySenderProvider
-                    .setClient(pubSubDownstreamSender(pubSubFactory, TelemetryConstants.TELEMETRY_ENDPOINT));
-            eventSenderProvider.setClient(pubSubDownstreamSender(pubSubFactory, EventConstants.EVENT_ENDPOINT));
+            final Optional<FixedCredentialsProvider> credentialsProvider = PubSubMessageHelper.getCredentialsProvider();
+            if (credentialsProvider.isPresent()) {
```

Comment:
Isn't it en error condition if we cannot create the *CredentialsProvider*? If so, we should at least log an error and/or throw an exception here to prevent (useless) startup ...